autoload/neomake/quickfix.vim

```diff
@@ -0,0 +1,192 @@
+" vim: ts=4 sw=4 et
+scriptencoding utf-8
+
+let s:maker_match_id = 997
+let s:gutter_match_id = 998
+let s:cursor_match_id = 999
+
+
+function! neomake#quickfix#enable() abort
+    let g:_neomake_qf_enabled = 1
+    augroup neomake_qf
+        autocmd!
+        autocmd FileType qf call neomake#quickfix#FormatQuickfix()
+    augroup END
+endfunction
+
+
+function! neomake#quickfix#disable() abort
+    let g:_neomake_qf_enabled = 0
+endfunction
+
+
+function! s:cursor_moved() abort
+    if b:neomake_start_col
+        if col('.') <= b:neomake_start_col + 1
+            call cursor(line('.'), b:neomake_start_col + 2)
+        endif
+
+        silent! call matchdelete(s:cursor_match_id)
+        call matchaddpos('neomakeCursorListNr',
+                    \ [[line('.'), (b:neomake_start_col - b:neomake_number_len) + 2, b:neomake_number_len]],
+                    \ s:cursor_match_id,
+                    \ s:cursor_match_id)
+    endif
+endfunction
+
+
+function! s:reset(buf) abort
+    call neomake#signs#ResetFile(a:buf)
+    execute 'sign unplace * buffer='.a:buf
+    call clearmatches()
+endfunction
+
+
+function! neomake#quickfix#FormatQuickfix() abort
+    if !get(g:, '_neomake_qf_enabled', 0) || &filetype != 'qf'
+        if exists('b:neomake_qf')
+            call s:reset(bufnr('%'))
+            unlet! b:neomake_qf
+            augroup neomake_qf
+                autocmd! * <buffer>
+            augroup END
+        endif
+        return
+    endif
+
+    let buf = bufnr('%')
+    call s:reset(buf)
+
+    let src_buf = 0
+    let loclist = 1
+    let qflist = getloclist(0)
+    if empty(qflist)
+        let loclist = 0
+        let qflist = getqflist()
+    endif
+
+    if empty(qflist) || qflist[0].text !~# '{neomake:[^}]\+}$'
+        set syntax=qf
+        return
+    endif
+
+    if loclist
+        let b:neomake_qf = 'file'
+        let src_buf = qflist[0].bufnr
+    else
+        let b:neomake_qf = 'project'
+    endif
+
+    runtime! syntax/neomake/qf.vim
+    if src_buf
+        let ft = getbufvar(src_buf, '&filetype')
+        if ft != ''
+            for name in split(ft, '\.')
```

Comment:
Alright so, that means the maker names should be truncated later.  Otherwise, there's no way to get that information.

If it's alright, I'm going to stick with just using the maker names for now and deal with this if issues arise from it.